testing/dart/color_test.dart

```diff
@@ -41,4 +45,9 @@ void main() {
       expect(e != null, equals(true));
     }
   });
+
+  test("two colors are only == if they have the same runtime type", () {
+    expect(const Color(123), isNot(equals(const NotAColor(123))));
```

Comment:
by which i meant, NotAColor == Color
But I guess we should also check NotAColor(12) == NotAColor(12)